diff --git a/BUILD.gn b/BUILD.gn
index 6e965691e..21c9d17ad 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1453,6 +1453,20 @@ angle_static_library("libGLESv2_static") {
   public_deps = [ ":libANGLE" ]
 }
 
+angle_static_library("libANGLE_static") {
+  sources = libglesv2_sources
+  sources += libegl_sources
+  defines = [
+    "ANGLE_STATIC=1", 
+    "ANGLE_EXPORT=",
+    "EGLAPI=",
+    "GL_APICALL=",
+    "GL_API="
+  ]
+  complete_static_lib = true
+  public_deps = [ ":libANGLE", "//build/config:shared_library_deps" ] 
+}
+
 angle_shared_library("libGLESv1_CM") {
   sources = libglesv1_cm_sources
   output_name = "libGLESv1_CM${angle_libs_suffix}"
diff --git a/include/platform/PlatformMethods.h b/include/platform/PlatformMethods.h
index 48ef33579..74d220066 100644
--- a/include/platform/PlatformMethods.h
+++ b/include/platform/PlatformMethods.h
@@ -15,6 +15,8 @@
 
 #define EGL_PLATFORM_ANGLE_PLATFORM_METHODS_ANGLEX 0x6AFB
 
+#define ANGLE_PLATFORM_EXPORT
+
 #if !defined(ANGLE_PLATFORM_EXPORT)
 #    if defined(_WIN32)
 #        if !defined(LIBANGLE_IMPLEMENTATION)
@@ -265,7 +267,7 @@ inline void DefaultPlaceholderCallback(...) {}
 
 #define ANGLE_PLATFORM_METHOD_DEF(Name, CapsName) CapsName##Func Name = Default##CapsName;
 
-struct ANGLE_PLATFORM_EXPORT PlatformMethods
+struct PlatformMethods
 {
     inline PlatformMethods();
 
diff --git a/src/libANGLE/renderer/vulkan/vk_utils.cpp b/src/libANGLE/renderer/vulkan/vk_utils.cpp
index b3c82efca..8b1cf835d 100644
--- a/src/libANGLE/renderer/vulkan/vk_utils.cpp
+++ b/src/libANGLE/renderer/vulkan/vk_utils.cpp
@@ -1219,6 +1219,7 @@ void InitExtendedDynamicState2EXTFunctions(VkDevice device)
 void InitFragmentShadingRateKHRInstanceFunction(VkInstance instance)
 {
     GET_INSTANCE_FUNC(vkGetPhysicalDeviceExternalSemaphorePropertiesKHR);
+    GET_INSTANCE_FUNC(vkGetPhysicalDeviceFragmentShadingRatesKHR);
 }
 
 void InitFragmentShadingRateKHRDeviceFunction(VkDevice device)
diff --git a/src/libEGL/libEGL_autogen.cpp b/src/libEGL/libEGL_autogen.cpp
index be4ce7716..3952c4bad 100644
--- a/src/libEGL/libEGL_autogen.cpp
+++ b/src/libEGL/libEGL_autogen.cpp
@@ -51,7 +51,7 @@ void EnsureEGLLoaded()
     }
 }
 #else
-void EnsureEGLLoaded() {}
+static inline void EnsureEGLLoaded() {}
 #endif  // defined(ANGLE_USE_EGL_LOADER)
 }  // anonymous namespace
 
